packages/rolldown/src/cli/index.ts

```diff
@@ -1,54 +1,33 @@
 import process from 'node:process'
-import parseArgs from 'mri'
 import { defineCommand, runMain, showUsage } from 'citty'
-import { bundle } from './commands/bundle'
+import { bundleWithCliOptions, bundleWithConfig } from './commands/bundle'
 import {
   version,
   description,
 } from '../../package.json' assert { type: 'json' }
-import { DEFAULT_CONFIG_FILENAME } from './constants'
-
-interface ParsedArgs {
-  config?: string | true
-  c?: string | true
-  // `citty` intercept the help option, so we don't need to deal with it
-  // help?: boolean
-  // h?: boolean
-}
+import { parseCliArguments } from './arguments'
 
 const main = defineCommand({
   meta: {
     name: 'rolldown',
     version,
     description,
   },
-  args: {
-    config: {
-      type: 'string',
-      alias: 'c',
-      description:
-        'Use this config file (if argument is used but value is unspecified, defaults to `rolldown.config.js`)',
-    },
-    help: {
-      type: 'boolean',
-      alias: 'h',
-      description: 'Show this help message',
-    },
-  },
-  async run(_ctx) {
-    // FIXME: `citty` doesn't support detecting if an argument is unspecified
-    const parsedArgs = parseArgs<ParsedArgs>(process.argv.slice(2))
-    let argConfig = parsedArgs.c || parsedArgs.config
-    if (argConfig) {
-      // If config is specified, we will ignore other arguments and bundle with the specified config
-      if (argConfig == true) {
-        argConfig = DEFAULT_CONFIG_FILENAME
-      }
-      await bundle(argConfig)
+  async run() {
+    const cliOptions = parseCliArguments()
+
+    if (cliOptions.config) {
+      await bundleWithConfig(cliOptions.config, cliOptions)
+      process.exit(0)
+    }
+
+    if ('input' in cliOptions.input) {
+      // If input is specified, we will bundle with the input options
+      await bundleWithCliOptions(cliOptions)
```

Comment:
Here, because the input argument is passed by the positional, so I think this might be the best way to identify if user input the input file in command line.

The command line will override the configure file if there's conflicts, in rollup.